webserver/static/js/feverscreen.ts

```diff
@@ -755,6 +907,124 @@ window.onload = async function() {
     return dest;
   }
 
+  function add_pixel(dest: Float32Array, x: number, y: number, amount: number, width: number, height: number) {
+    x=~~x
+    y=~~y
+    if(x<0 || y<0) { return;}
+    if(x>=width || y>=height) { return;}
+    const index = y * width + x;
+    dest[index] += amount;
+  }
+
+  function add_circle(dest: Float32Array, cx: number, cy: number, radius: number, amount: number, width: number, height: number) {
```

Comment:
Why not use the overlay canvas?

Reply:
This is the Hough Circle Transform, which uses additive blending, and potentially more than an 8 bit frame buffer, with a final framebuffer read at the end, once for each radius scanned. Adapting it to use HTML/Canvas is both subtle and potentially slower.

Reply:
We have a function called `drawTargetCircle` and a function called `addCircle` now though. I think it would be better to just have one function, something like `drawCircle(x, y, radius, colour, fill)`

Reply:
drawTargetCircle uses the Canvas API to defer a "Target Circle" onto a canvas. The circle need not even be a strict circle, it could be decorated with tick marks or guide marks for example, or even be a cross hairs. Similarly the color is not specified in the function definition, as it should come from the CSS, or could even be multicolored or have a gradient.

OTOTH, addCircle modifies an accumulation buffer, and literally adds the image of a circle that is precisely 1.0 * Pi * radius^2 units worth of weight into that accumulation buffer. There is no color as the destination is a floating point single channel accumulation buffer.
addCircle is used as part of the Hough Circle Transform.

Reply:
OK so then it needs to have a different name than `addCircle` or be in a file for the circle detection or have a description for what the function is doing.
I think you should separate out the thermal reference detection as you did with the haar cascade. This makes it much easier to review.

Reply:
Fixed in a49fd70